packages/frontend/src/state/lp/hooks.ts

```diff
@@ -0,0 +1,115 @@
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk'
+import { fromTokenAmount } from '@utils/calculations'
+import { useAtomValue } from 'jotai'
+import { addressesAtom, isWethToken0Atom } from '../positions/atoms'
+import BigNumber from 'bignumber.js'
+import { OSQUEETH_DECIMALS } from '@constants/index'
+import { controllerContractAtom, controllerHelperHelperContractAtom, nftManagerContractAtom, quoterContractAtom, squeethPoolContractAtom } from '../contracts/atoms'
+import useAppCallback from '@hooks/useAppCallback'
+import { addressAtom } from '../wallet/atoms'
+import { Contract } from 'web3-eth-contract'
+import { useHandleTransaction } from '../wallet/hooks'
+import { ethers } from 'ethers'
+import { useCallback } from 'react'
+import { useGetDebtAmount, useGetTwapSqueethPrice, useGetVault } from '../controller/hooks'
+
+/*** GETTERS ***/
+
+export const useGetPosition = () => {
+    const contract = useAtomValue(nftManagerContractAtom)
+  
+    const getPosition = useCallback(
+      async (uniTokenId: number) => {
+        if (!contract) return null
+        const position = await contract.methods.positions(uniTokenId).call()
+        const { nonce, operator, token0, token1, fee, tickLower, tickUpper, liquidity, feeGrowthInside0LastX128, feeGrowthInside1LastX128, tokensOwed0, tokensOwed1 } = position
+        return {
+          nonce,
+          operator,
+          token0,
+          token1,
+          fee,
+          tickLower,
+          tickUpper,
+          liquidity,
+          feeGrowthInside0LastX128,
+          feeGrowthInside1LastX128,
+          tokensOwed0,
+          tokensOwed1
+        }
+      },
+      [contract],
+    )
+  
+    return getPosition
+  }
+
+  export const useGetDecreaseLiquidity = () => {
+    const contract = useAtomValue(nftManagerContractAtom)
+  
+    const getDecreaseLiquiduity = useCallback(
+      async (tokenId: number, liquidity: number, amount0Min: number, amount1Min: number, deadline: number) => {
+        if (!contract) return null
+        const DecreaseLiquidityParams = {
+          tokenId,
+          liquidity,
+          amount0Min,
+          amount1Min,
+          deadline,
+        }
+        console.log("DecreaseLiquidityParams", DecreaseLiquidityParams)
+  
+        const decreaseLiquidity = await contract.methods.decreaseLiquidity(DecreaseLiquidityParams).call()
```

Comment:
what is this function used for/do?  is this called on the NonFungiblePositionManager contract?

Reply:
I guess I don't really undestand what is going on here. This is a write function right? Can we call it with a random address like this?  A user won't hold the NFT, so won't it revert?

Reply:
This is how we got the amount0 and amount1 in the ControllerHelper tests as well. It's just a static call (does not change contract state), which means it functions as view (not write).

There's other ways to get amount0 and amount1 with things like the `LiquidityAmounts.sol` contract, but this is the simplest.

Reply:
We could calculate from liquidty and ticks too but this is probably more precise